### PR TITLE
2198 tei editor

### PR DESCRIPTION
**Pull request by benwbrum**
This adds the following URLs
* `/feature/codemirror/enable` enables the TEI editor
* `/feature/codemirror/disable` disables the TEI editor
* `/feature/codemirror` displays current status of the TEI editor
* `/feature/line_numbers/enable` enables line number display in the TEI editor
* `/feature/line_numbers/enable` disables line number display in the TEI editor
* `/feature/line_numbers` displays current status of line numbers within the editor

It also adds display support for tags used within the [CWG-K stylesheet](https://github.com/CivilWarGovernorsOfKentucky/Omeka/blob/fe414acc822f411bbd21c4bc897e2e349b37132a/files/xslt/content/testCWG.xsl)

### CODE REVIEW

File: app/assets/config/manifest.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
 //= link_tree ../images
 //= link_directory ../javascripts .js
 //= link_directory ../stylesheets .css
+//= link_directory ../stylesheets/codemirror/lib .css
+//= link_directory ../stylesheets/codemirror-buttons .css
```
Comment (saracarl) [inline app/assets/config/manifest.js:5]: This is weird -- the space before the .css -- maybe it's just github?

File: app/controllers/user_controller.rb [hunk 2]
```diff
@@ -8,6 +8,30 @@ def demo
...
+      end
+      return
+    end
+    redirect_back :fallback_location => dashboard_role_path
+  end
+
+  def disable_codemirror
+    Flipper.disable(:codemirror)
+    redirect_back :fallback_location => dashboard_role_path
+  end
```
Comment (saracarl) [inline app/controllers/user_controller.rb:33]: I don't see this method being called anywhere.

File: config/initializers/flipper.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'flipper'
+
```
Comment (saracarl) [inline config/initializers/flipper.rb:2]: We either need an updated Gemfile with flipper in it or this file shouldn't be here.

**Review by saracarl — CHANGES_REQUESTED** [review]

File: app/assets/config/manifest.js (hunk 1 above)
Comment (saracarl) [reply app/assets/config/manifest.js:5]: nope, definitely is a problem on my filesystem as well.

Comment (benwbrum) [reply app/assets/config/manifest.js:5]: I don't think this is a problem.

Comment (saracarl) [reply app/assets/config/manifest.js:5]: Discussed and decided this was OK.

**Pull request merged**